File: Tests/SwiftKotlinFrameworkTests/Tests/KotlinTokenizer/annotations.swift

// escaping
var completionHandlers: [() -> Void] = []
func someFunctionWithEscapingClosure(completionHandler: @escaping () -> Void) {
    completionHandlers.append(completionHandler)
}

func serve(customer customerProvider: @autoclosure () -> String) {
    print("Now serving \(customerProvider())!")
}

func collectCustomerProviders(_ customerProvider: @autoclosure @escaping () -> String) {
    customerProviders.append(customerProvider)
}

@discardableResult
func foo(_ code:(() -> String)) -> String {
    return "foo \(bar(code))"
}

File: Assets/Tests/KotlinTokenizer/annotations.kt
internal var completionHandlers: List<() -> Unit> = listOf()

internal fun someFunctionWithEscapingClosure(completionHandler: () -> Unit) {
    completionHandlers.append(completionHandler)
}

internal fun serve(customerProvider: () -> String) {
    print("Now serving ${customerProvider()}!")
}

internal fun collectCustomerProviders(customerProvider: () -> String) {
    customerProviders.append(customerProvider)
}

internal fun foo(code: (() -> String)) : String =
    "foo ${bar(code)}"
